sfs/time/drivingfunction.py

```diff
@@ -2,6 +2,34 @@
 
 .. include:: math-definitions.rst
 
+.. plot::
+    :context: reset
+
+    import matplotlib.pyplot as plt
+    import numpy as np
+    import sfs
+
+    plt.rcParams['figure.figsize'] = 6, 6
+
+    xs = -1.5, 1.5, 0  # point source
+    npw = sfs.util.direction_vector(np.radians(-45))  # plane wave
+    xf = -0.5, 0.5, 0  # focused source
+    nf = sfs.util.direction_vector(np.radians(-45))  # normal vector for focused source
+
+    # Impulsive excitation
+    signal = sfs.util.DelayedSignal(np.insert(np.zeros(511), 0, 1), 44100, 0)
```

Comment:
This is not at all how this is meant to be used!

As the documentation (http://python.sfstoolbox.org/en/0.4.0/utilities.html#sfs.util.DelayedSignal) says:

> This class (a `collections.namedtuple`) is not meant to be instantiated by users.
>
> To pass a signal to a function, just use a simple `tuple` or `list` containing the audio data and the sampling rate (in Hertz), with an optional starting time (in seconds) as a third item.

The future goal for using this is like that:

```python
signal = [1], 44100
```

... but sadly, this doesn't work yet because the resulting signals are cropped. We still have to come up with a solution to that.

For now, it should suffice to add a few zeros:

```python
signal = [1, 0, 0, 0], 44100
```

Do you really need that many zeros?

You can of course also use a NumPy array, but you should not use `DelayedSignal`.

Reply:
`as_delayed_signal` shouldn't be used, either?

We need more zeros for NFC-HOA driving signals.

Reply:
> `as_delayed_signal` shouldn't be used, either?

Not in this case, no.
If you create a signal in a script, it is sufficient and preferable to simply use a `tuple` (or a `list`).

The helper function `as_delayed_signal()` is meant to be used inside a function that takes a signal as an input.

Basically every function in `sfs.time` that takes a `signal` argument will use `as_delayed_signal()` (unless it directly forwards the `signal` argument to another function which in turn will use `as_delayed_signal()`).

> We need more zeros for NFC-HOA driving signals.

OK, then it makes sense to use `np.zeros()` at some point. I find the `np.insert()` thing a bit confusing, but I guess it's OK.

Reply:
The unit impulse is now generated by [`scipy.signal.unit_impulse()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.unit_impulse.html).

Reply:
Cool, I didn't even know such a function existed, very handy!